templates/plugin-gruntfile.mustache

```diff
@@ -11,10 +11,11 @@ module.exports = function( grunt ) {
 			options: {
 				textdomain: '{{textdomain}}',
 			},
-			target: {
-				files: {
-					src: [ '*.php', '**/*.php', '!node_modules/**', '!php-tests/**', '!bin/**' ]
-				}
+			update_all_domains: {
```

Comment:
Given the following sample code, I'll list the results depending on each config.

```
$label = __( 'Hello World', '{{plugin_slug}}' );
$label = __( 'Hello World', 'other-plugin-slug' );
$label = __( 'Hello World' );
```

With `addtextdomain.target`:

```
$label = __( 'Hello World', '{{plugin_slug}}' );
$label = __( 'Hello World', 'other-plugin-slug', '{{plugin_slug}}' );
$label = __( 'Hello World', '{{plugin_slug}}' );
```

With `addtextdomain.update_all_domains.updateDomains = true`:

```
$label = __( 'Hello World', '{{plugin_slug}}' );
$label = __( 'Hello World', '{{plugin_slug}}' );
$label = __( 'Hello World', '{{plugin_slug}}' );
```

And with `addtextdomain.update_all_domains.updateDomains = '{{plugin_slug}}'`:

```
$label = __( 'Hello World', '{{plugin_slug}}' );
$label = __( 'Hello World', 'other-plugin-slug', '{{plugin_slug}}' );
$label = __( 'Hello World', '{{plugin_slug}}' );
```

I thought I saw different results when I was testing earlier.

So it seems that the only issue that exists is that if you happened to copy code that uses a different textdomain, your domain will get appended to the `i18n` function(which is definitely not correct).

Not sure if the recommended should be forcing all textdomains to be the same by default. Maybe this change is not really necessary?